ptclient/ldap.c

```diff
@@ -1429,12 +1429,13 @@ static int ptsmodule_make_authstate_group(
         entry = ldap_first_entry(ptsm->ld, res2);
         char **uvals = ldap_get_values(ptsm->ld, entry, (char *)ptsm->user_attribute);
         int unumvals = ldap_count_values(uvals);
-        if (!unumvals) continue;  // no user attribute on the DN
-
-        // create a group item
-        strlcat((*newstate)->groups[j].id, uvals[0], sizeof((*newstate)->groups[j].id));
-        (*newstate)->groups[j].hash = strhash((*newstate)->groups[j].id);
-        j++;
+        int k;
+        for (k = 0; k < unumvals; k++) {
+            // create a group item
+            strlcat((*newstate)->groups[j].id, uvals[k], sizeof((*newstate)->groups[j].id));
```

Comment:
Hmm.  Do we need a separator of some sort between uvals here?  I don't follow exactly what the generated id is used for, but it smells like there could be a problem if "foo", "bar", and "foobar" are all valid user_attribute values and the record contains both "foo" and "bar".

Perhaps it makes more sense to mandate that there's only one.  I don't think that would be surprising in the LDAP world, it's normal for a schema to constrain particular attributes to be single-valued, multi-valued, etc.

Reply:
Ok, this is getting a bit into the very very weeds.  if you look a couple of lines later, there's a `j++`, so maybe this should be strncpy instead but no, that's not an issue.  We're adding all the items.  This does mean that a user might be added more than once but that's already an issue from this half-arsed underlying structure.

Reply:
Oh right, yeah, I missed how `j` was being used here, my bad

Reply:
@elliefm are you happy to approve now?  Hoping to get this one merged so we can build on top :)